e2e/framework/addon/eso_argocd_application.go

```diff
@@ -37,67 +41,84 @@ type ArgoCDApplication struct {
 	HelmValues           string
 
 	config *Config
+	dc     *dynamic.DynamicClient
 }
 
+var argoApp = schema.GroupVersionResource{
+	Group:    "argoproj.io",
+	Version:  "v1alpha1",
+	Resource: "applications",
+}
```

Comment:
note: the dependency to argocd blocks the upgrade, hence i decided to deploy argo `Kind=Application` the `unstructured` way.